### PR TITLE
[aws-lambda] Multi-value headers and query params in Api Gateway and Authorizer events

### PR DESCRIPTION
**Pull request by trevor-leach**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#apigateway-multivalue-headers-and-parameters
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/aws-lambda/index.d.ts [hunk 1]
```diff
@@ -75,8 +78,10 @@ export interface CustomAuthorizerEvent {
     methodArn: string;
     authorizationToken?: string;
     headers?: { [name: string]: string };
+    multiValueHeaders?: { [name: string]: string[] };
     pathParameters?: { [name: string]: string } | null;
     queryStringParameters?: { [name: string]: string } | null;
+    multiValueQueryStringParameters?: { [name: string]: string[] } | null;
```
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts]: Just noticed these inconsistent `| null` here. Gross, as all these properties are all certainly going to be one of not present or `null` if they even are optional.
That said, not your problem!

**Review by simonbuchan — APPROVED** [review]
Great, thanks for this! (I certainly have wanted this feature in the past!)

**Comment by simonbuchan** [comment]
This only has the request multi-value headers, would you also like to add the response multi-value headers mentioned in the link? That would be super handy for `Set-Cookie`...

**Comment by trevor-leach** [comment]
@simonbuchan I fixed a merge conflict and added multiValueHeaders to the response.

**Comment by DanielRosenwasser** [comment]
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

If you don't care about any of this stuff, I can always just pull in the changes.

**Comment by trevor-leach** [comment]
Thanks for the heads-up.  I think I've set my github account's email and ammended the last commit.  If  not, please go ahead and pull the changes anyway.

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**